src/integration/resources/inprocess/dbnode_test.go

```diff
@@ -0,0 +1,201 @@
+// +build integration_v2
+// Copyright (c) 2021  Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package inprocess
+
+import (
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/m3db/m3/src/dbnode/generated/thrift/rpc"
+	"github.com/m3db/m3/src/integration/resources"
+	"github.com/m3db/m3/src/m3ninx/idx"
+	"github.com/m3db/m3/src/query/generated/proto/admin"
+	"github.com/m3db/m3/src/x/checked"
+	"github.com/m3db/m3/src/x/ident"
+	"github.com/m3db/m3/src/x/pool"
+	"github.com/m3db/m3/src/x/serialize"
+)
+
+func TestNewDBNode(t *testing.T) {
+	_, closer := setupNode(t)
+	closer()
+}
+
+func TestHealth(t *testing.T) {
+	dbnode, closer := setupNode(t)
+	defer closer()
+
+	res, err := dbnode.Health()
+	require.NoError(t, err)
+
+	require.True(t, res.Ok)
+}
+
+func TestWaitForBootstrap(t *testing.T) {
+	dbnode, closer := setupNode(t)
+	defer closer()
+
+	res, err := dbnode.Health()
+	require.NoError(t, err)
+
+	require.Equal(t, true, res.Bootstrapped)
+}
+
+func TestWriteFetchRoundtrip(t *testing.T) {
+	dbnode, closer := setupNode(t)
+	defer closer()
+
+	var (
+		id  = "foo"
+		ts  = time.Now()
+		val = 1.0
+	)
+	req := &rpc.WriteRequest{
+		NameSpace: resources.UnaggName,
+		ID:        id,
+		Datapoint: &rpc.Datapoint{
+			Timestamp: ts.Unix(),
+			Value:     val,
+		},
+	}
+	require.NoError(t, dbnode.WritePoint(req))
+
+	freq := &rpc.FetchRequest{
+		RangeStart: ts.Add(-1 * time.Minute).Unix(),
+		RangeEnd:   ts.Add(1 * time.Minute).Unix(),
+		NameSpace:  resources.UnaggName,
+		ID:         id,
+	}
+	res, err := dbnode.Fetch(freq)
+	require.NoError(t, err)
+
+	require.Equal(t, 1, len(res.Datapoints))
+	require.Equal(t, rpc.FetchResult_{
+		Datapoints: []*rpc.Datapoint{
+			{Timestamp: ts.Unix(), Value: val},
+		},
+	}, res.Datapoints[0])
+}
+
+func TestWriteTaggedFetchTaggedRoundtrip(t *testing.T) {
```

Comment:
One interesting thing you can see from this test is that, while node setup is pretty painless, reading and writing could be less frictionless. Adding a layer of helpers after we reach dtest parity could be really nice here.